browser/blueprints.py

```diff
@@ -249,82 +159,179 @@ def form():
-    Request HTML caliban tool page to be rendered if user makes a URL
+    Request HTML DeepCell Label tool page to be rendered if user makes a URL
     request to access a specific data file that has been preloaded to the
     input S3 bucket (ex. http://127.0.0.1:5000/test.npz).
     """
-    settings = make_settings(filename)
-    return render_template(
-        'tool.html',
-        settings=settings)
 
-
-def get_edit(project):
-    """Factory for Edit objects"""
-    filename = project.path
-    if is_npz_file(filename):
-        return ZStackEdit(project)
-    elif is_trk_file(filename):
-        # don't use RGB mode with track files
-        return TrackEdit(project)
-    return BaseEdit(project)
-
-
-def make_settings(filename):
-    """Returns a dictionary of settings to send to the front-end."""
     folders = re.split('__', filename)
-
     # TODO: better parsing when buckets are not present
     input_bucket = folders[0] if len(folders) > 1 else S3_INPUT_BUCKET
     output_bucket = folders[1] if len(folders) > 2 else S3_OUTPUT_BUCKET
     start_of_path = min(len(folders) - 1, 2)
     path = '__'.join(folders[start_of_path:])
 
-    rgb = request.args.get('rgb', default='false', type=str)
-    pixel_only = request.args.get('pixel_only', default='false', type=str)
-    label_only = request.args.get('label_only', default='false', type=str)
     # TODO: uncomment to use URL parameters instead of rigid bucket formatting within filename
     # input_bucket = request.args.get('input_bucket', default=S3_INPUT_BUCKET, type=str)
     # output_bucket = request.args.get('output_bucket', default=S3_OUTPUT_BUCKET, type=str)
 
-    if is_trk_file(filename):
+    return render_template(
+        'loading.html',
+        input_bucket=input_bucket,
+        output_bucket=output_bucket,
+        path=path)
+
+
+@bp.route('/api/project/<token>', methods=['GET'])
+def get_project(token):
+    """
+    Retrieve data from a project already in the Project table.
+    """
+    start = timeit.default_timer()
+    project = Project.get(token)
+    if not project:
+        return abort(404, description=f'project {token} not found')
+    # arg is 'false' which gets parsed to True if casting to bool
+    rgb = request.args.get('rgb', default='false', type=str)
+    rgb = bool(distutils.util.strtobool(rgb))
+    project.rgb = rgb
+    project.update()
+    payload = project.make_first_payload()
+    current_app.logger.debug('Loaded project %s in %s s.',
+                             project.token, timeit.default_timer() - start)
+    return jsonify(payload)
+
+
+@bp.route('/api/project', methods=['POST'])
+def create_project():
+    """
+    Create a new Project.
+    """
+    start = timeit.default_timer()
+    loader = loaders.get_loader(request)
+    project = Project.create(loader)
+    current_app.logger.info('Created project from %s in %s s.',
+                            loader.path, timeit.default_timer() - start)
+    return {'projectId': project.token}
```

Comment:
No jsonify required?

Reply:
I added jsonify, not sure how it worked without it. Maybe because it's such a simple payload it flew under the radar